### ISSUE TITLE
drc_cache: Error allocating virtual memory

### ISSUE THREAD

**Issue by FollyMaddy** [mamedev/mame#8274]
When using "mame -listdevices" or "mame -listmedia" mame always gives an error,
drc_cache: Error allocating virtual memory
It does output some good info first, but always stops halfway and at the same point.
I experience this on both the raspberry pi 4 and on an x86 with 32 bit buster.
I experience this issue from mame0228 up to mame0233.

On a x86-64 with Ubuntu 20.04 64 bit I have  had no issues with mame231.
I didn't try other mame version on the 64 bit OS.

It would be nice if this issue could be fixed as my script depends on extracting data from mame.

BTW. great work we love using it.

**Comment by angelosa** [comment]
It's fair to assume that your RPi OS is 32-bit since that left beta phase earlier this month.
While the report lacks **where exactly it crashes**, I wonder if appending `-nodrc`  will do the trick (that error is supposedly from the dynamic recompiler, which in context shouldn't really throw that).
We also dropped official 32-bit support in 2019, wontfix?

**Comment by cuavas** [comment]
`-nodrc` won’t stop it from allocating memory for the DRC cache when instantiating those devices.  I’m pretty sure we’re doing things properly, since it’s tested and working across multiple operating systems.  It’s more likely misconfiguration of a security policy preventing MAME from marking pages executable (AppArmor, SELinux, or similar).

**Comment by cuavas** [comment]
Yes, it needs to build machine configuration to do all those things, so for any CPU that has recompiler modes it will allocate pages and mark them executable.  Check that you don’t have SELinux, AppArmor, or something else configured to prevent this.

**Comment by smf-** [comment]
what if you do:

mame -listdevices jak_spdmoa

mame -listmedia jak_spdmo

If it's something about virtual memory then that will give the same
error, if it's a leak then it will be fine.

You could even try

mame -listdevices jak*

mame -listmedia jak*

You could also use something that monitors the mame memory usage when
using -listdevices / -listmedia. If it just keeps going up and up then
there is a leak.

**Comment by FollyMaddy** [comment]
SElinux seems not in use :
```
pi@raspberrypi:~ $ sestatus
bash: sestatus: command not found
```
AppArmor seems not in use
```
pi@raspberrypi:~ $ aa-status
apparmor module is loaded.
apparmor filesystem is not mounted.
```

> mame -listdevices jak_spdmoa
> mame -listmedia jak_spdmo
Both work.

>mame -listdevices jak*
>mame -listmedia jak*
Both these give the error too.

Well I fixed my script so it doesn't have to use these mame commands anymore.
So if we can't find the exact cause then this isn't a disaster.
It's probably better to stop searching.

Thank you all for the help.
